x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/setup_technology.ts

```diff
@@ -94,18 +98,43 @@ export function useSetupTechnology({
     isAgentlessCloudEnabled,
     isAgentlessServerlessEnabled,
   } = useAgentless();
+
   // this is a placeholder for the new agent-BASED policy that will be used when the user switches from agentless to agent-based and back
-  const [newAgentBasedPolicy] = useState<NewAgentPolicy | undefined>({ ...newAgentPolicy });
+  const newAgentBasedPolicy = useRef<NewAgentPolicy>(newAgentPolicy);
   const [selectedSetupTechnology, setSelectedSetupTechnology] = useState<SetupTechnology>(
     SetupTechnology.AGENT_BASED
   );
   const [newAgentlessPolicy, setNewAgentlessPolicy] = useState<AgentPolicy | NewAgentPolicy>(
     generateNewAgentPolicyWithDefaults({
-      name: `Agentless policy ${uuidv4()}`,
       supports_agentless: true,
     })
   );
 
+  useEffect(() => {
+    if (isEditPage) {
+      return;
+    }
+    if (isAgentlessCloudEnabled && selectedSetupTechnology === SetupTechnology.AGENTLESS) {
+      const nextNewAgentlessPolicy = {
+        ...newAgentlessPolicy,
+        name: getAgentlessAgentPolicyNameFromPackagePolicyName(packagePolicy.name),
+      };
+      if (nextNewAgentlessPolicy.name !== newAgentlessPolicy.name) {
+        setNewAgentlessPolicy(nextNewAgentlessPolicy);
+        updateNewAgentPolicy(nextNewAgentlessPolicy as NewAgentPolicy);
```

Comment:
If we are calling `updateNewAgentPolicy` and `updateAgentPolicies` here,  I don't think we need to [do it handleSetupTechnologyChange also](https://github.com/elastic/kibana/pull/187989/files#diff-0ab1bc414c990f143b0013058e8394e4c66889306da52ac55e58965ccc851283R171)?

If not, we may only need to set the selected tab in the handle in the `handleSetupTechnologyChange`

Reply:
Not really, this one intentionally only triggers when the AgentlessPolicy name changes (`nextNewAgentlessPolicy.name !== newAgentlessPolicy.name`), so just switching between agentless and agent-based without editing the integration name would not trigger `updateNewAgentPolicy` and `updateAgentPolicies`.